src/backend/cassandra/impl/Result.h

```diff
@@ -0,0 +1,265 @@
+        output = ripple::uint256::fromVoid(buf);
+    }
+    else if constexpr (std::is_same_v<decayed_t, ripple::AccountID>)
+    {
+        cass_byte_t const* buf;
+        std::size_t bufSize;
+        auto const rc =
+            cass_value_get_bytes(cass_row_get_column(row, idx), &buf, &bufSize);
+        throwErrorIfNeeded(rc, "Extract ripple::AccountID");
+        output = ripple::AccountID::fromVoid(buf);
+    }
+    else if constexpr (std::is_same_v<decayed_t, uchar_vector_t>)
+    {
+        cass_byte_t const* buf;
+        std::size_t bufSize;
+        auto const rc =
+            cass_value_get_bytes(cass_row_get_column(row, idx), &buf, &bufSize);
+        throwErrorIfNeeded(rc, "Extract vector<unsigned char>");
+        output = uchar_vector_t{buf, buf + bufSize};
+    }
+    else if constexpr (std::is_same_v<decayed_t, uint_tuple_t>)
+    {
+        auto const* tuple = cass_row_get_column(row, idx);
+        output = TupleIterator::fromTuple(tuple).extract<uint32_t, uint32_t>();
+    }
+    else if constexpr (std::is_convertible_v<decayed_t, std::string>)
+    {
+        char const* value;
+        std::size_t len;
+        auto const rc =
+            cass_value_get_string(cass_row_get_column(row, idx), &value, &len);
+        throwErrorIfNeeded(rc, "Extract string");
+        output = std::string{value, len};
+    }
+    else if constexpr (std::is_same_v<decayed_t, bool>)
+    {
+        cass_bool_t flag;
+        auto const rc =
+            cass_value_get_bool(cass_row_get_column(row, idx), &flag);
+        throwErrorIfNeeded(rc, "Extract bool");
+        output = flag ? true : false;
+    }
+    // clio only uses bigint (int64_t) so we convert any incoming type
+    else if constexpr (std::is_convertible_v<decayed_t, int64_t>)
+    {
+        int64_t out;
+        auto const rc =
+            cass_value_get_int64(cass_row_get_column(row, idx), &out);
+        throwErrorIfNeeded(rc, "Extract int64");
+        output = static_cast<decayed_t>(out);
+    }
+    else
+    {
+        // type not supported for extraction
+        static_assert(unsupported_v<decayed_t>);
+    }
+
+    return output;
+}
+
+struct Result : public ManagedObject<CassResult const>
+{
+    /* implicit */ Result(CassResult const* ptr);
+
+    [[nodiscard]] std::size_t
+    numRows() const;
+
+    [[nodiscard]] bool
+    hasRows() const;
+
+    template <typename... RowTypes>
+    std::optional<std::tuple<RowTypes...>>
+    get() const requires(std::tuple_size<std::tuple<RowTypes...>>{} > 1)
+    {
+        // row managed internally by cassandra driver, hence no ManagedObject.
+        auto const* row = cass_result_first_row(*this);
+        if (row == nullptr)
+            return std::nullopt;
+
+        std::size_t idx = 0;
+        auto advanceId = [&idx]() { return idx++; };
+
+        return std::make_optional<std::tuple<RowTypes...>>(
+            {extractColumn<RowTypes>(row, advanceId())...});
```

Comment:
can we use idx++ directly?

Reply:
We can and it works in practice.. however, unfortunately, the compiler complains that the order may be undefined.. and with `-Werror` it fails to compile on gcc. So i had to add this extra step - luckily it does not actually cost anything at runtime.